src/bitcoin-cli.cpp

```diff
@@ -222,12 +223,18 @@ UniValue CallRPC(const std::string& strMethod, const UniValue& params)
         // Try fall back to cookie-based authentication if no password is provided
         if (!GetAuthCookie(&strRPCUserColonPass)) {
             throw std::runtime_error(strprintf(
-                _("Could not locate RPC credentials. No authentication cookie could be found, and no rpcpassword is set in the configuration file (%s)"),
-                    GetConfigFile(GetArg("-conf", BITCOIN_CONF_FILENAME)).string().c_str()));
-
+                _("Could not locate RPC credentials. No authentication cookie could be found, and RPC password is not set.  See -rpcpassword and -stdinrpcpass.  Configuration file: (%s)"),
+                GetConfigFile(GetArg("-conf", BITCOIN_CONF_FILENAME)).string().c_str()));
         }
     } else {
-        strRPCUserColonPass = GetArg("-rpcuser", "") + ":" + GetArg("-rpcpassword", "");
+        strRPCUserColonPass = GetArg("-rpcuser", strRPCUserColonPass);
+        if (strRPCUserColonPass.empty()) {
+            throw std::runtime_error(strprintf(
+                _("Could not locate RPC credentials. No authentication cookie could be found, and RPC user is not set.  See -rpcuser.  Configuration file: (%s)"),
```

Comment:
I don't understand this change. I think it's better to leave it out as it's not relevant to adding `-stdinrpcpass`, it also duplicates the credentials error which is a bit ugly.